attest/untrusted/src/sim.rs

```diff
@@ -302,4 +318,125 @@ mod test {
         uut.clear();
         assert_eq!(DcapEvidence::default(), uut);
     }
+
+    #[test]
+    fn default_dcap_evidence_digest_is_not_an_empty_digest() {
+        let empty_evidence = DcapEvidence::default();
+        let digest = empty_evidence.digest32::<MerlinTranscript>(b"");
+
+        let mut empty_digest = [0u8; 32];
+        let transcript: MerlinTranscript = DigestTranscript::new();
+        transcript.extract_digest(&mut empty_digest);
+
+        assert_ne!(digest, empty_digest);
+    }
+
+    #[test]
+    fn dcap_evidence_digest() {
+        let (report, report_data) = report_and_report_data();
+        let quote = SimQuotingEnclave::quote_report(&report).expect("Failed to get quote");
+        let tcb_info = SIM_TCB_INFO.to_owned();
+        let collateral = collateral_from_from_raw_data(
+            &tcb_info,
+            SIM_QE_IDENTITY,
+            SIM_CRL,
+            IAS_SIM_SIGNING_CHAIN,
+        );
+
+        let evidence_1 = DcapEvidence {
+            quote: Some(quote.clone()),
+            collateral: Some(collateral.clone()),
+            report_data: Some(report_data.clone()),
+        };
+
+        let evidence_2 = DcapEvidence {
+            quote: Some(quote.clone()),
+            collateral: Some(collateral.clone()),
+            report_data: Some(report_data.clone()),
+        };
+
+        let digest_1 = evidence_1.digest32::<MerlinTranscript>(b"");
+        let digest_2 = evidence_2.digest32::<MerlinTranscript>(b"");
+        assert_eq!(digest_1, digest_2);
+
+        let mut quote_bytes: Vec<u8> = quote.as_ref().to_vec();
+        // Need to jump past the `Version` and `Attestation Type` bytes or else
+        // building a quote will fail.
+        quote_bytes[4] += 1;
+        let modified_quote =
+            Quote3::try_from(quote_bytes).expect("Failed to create modified quote");
+        let modified_quote_evidence = DcapEvidence {
+            quote: Some(modified_quote),
+            collateral: Some(collateral.clone()),
+            report_data: Some(report_data.clone()),
+        };
+
+        let modified_quote_digest = modified_quote_evidence.digest32::<MerlinTranscript>(b"");
+        assert_ne!(digest_1, modified_quote_digest);
```

Comment:
Similar to #3554 not fond of this workflow style test, but couldn't come up with a nicer way.